packages/node_modules/@webex/internal-plugin-ediscovery/src/transforms.js

```diff
@@ -0,0 +1,351 @@
+          ctx.webex.logger.error(`Error decrypting space name for report ${object.body.id}: ${reason}`);
+        });
+    }
+
+    let keywordPromises = [];
+
+    if (reportRequest.keywords) {
+      keywordPromises = Promise.all(reportRequest.keywords.map((keyword) => ctx.webex.internal.encryption.decryptText(reportRequest.encryptionKeyUrl, keyword)))
+        .then((decryptedKeywords) => {
+          reportRequest.keywords = decryptedKeywords;
+        })
+        .catch((reason) => {
+          ctx.webex.logger.error(`Error decrypting keywords for report ${object.body.id}: ${reason}`);
+        });
+    }
+
+    let emailPromises = [];
+
+    if (reportRequest.emails) {
+      emailPromises = Promise.all(reportRequest.emails.map((email) => ctx.webex.internal.encryption.decryptText(reportRequest.encryptionKeyUrl, email)))
+        .then((decryptedEmails) => {
+          reportRequest.emails = decryptedEmails;
+        })
+        .catch((reason) => {
+          ctx.webex.logger.error(`Error decrypting emails for report ${object.body.id}: ${reason}`);
+        });
+    }
+
+    return Promise.all([reportNamePromise, reportDescriptionPromise].concat(spaceNamePromises, keywordPromises, emailPromises));
+  }
+
+  static decryptReportContent(ctx, object, reportId) {
+    if (!object || !object.body || !reportId) {
+      return Promise.resolve();
+    }
+    const activity = object.body;
+
+    const promises = [];
+
+    return ctx.webex.internal.ediscovery.getContentSummaryBySpaceId(reportId, activity.targetId)
+      .then((space) => {
+        if (!space) {
+          const reason = `Space ${activity.targetId} may not contain any users from this organisation - unable to decrypt activity ${activity.activityId}`;
+
+          activity.error = reason;
+          ctx.webex.logger.error(reason);
+
+          return Promise.resolve(object);
+        }
+
+        if (space.error) {
+          // add error property to activity - this error will be recorded in the downloader
+          activity.error = space.error;
+        }
+
+        // set space name and participants on activity
+        if (space.spaceName) {
+          activity.spaceName = space.spaceName;
+        }
+        else if (space.isOneOnOne && space.participantDisplayNames && space.participantDisplayNames.length === 2) {
+          // One to One spaces have no space name, use participant names as 'Subject' instead
+          activity.spaceName = space.participantDisplayNames.toString().replace(',', ' & ');
+        }
+        else {
+          activity.spaceName = '';
+        }
+
+        if (space.participantDisplayNames) {
+          // Remove sender from list of participants as they will appear in the 'To' field
+          activity.participantDisplayNames = space.participantDisplayNames.filter((value) => value !== activity.actorDisplayName).toString();
+        }
+
+        let onBehalfOfUser;
+
+        if (space.onBehalfOfUser) {
+          onBehalfOfUser = space.onBehalfOfUser;
```

Comment:
This is the new change to fix some decryption issues we were seeing around empty spaces, external users etc. We were just picking the first uuid in the list of participants, now the service determines which one we should use.